symfony/plugins/orangehrmTimePlugin/Dao/ProjectActivityDao.php

```diff
@@ -173,4 +173,54 @@ public function isProjectActivityNameTaken(int $projectId, string $projectActivi
         }
         return $this->getPaginator($q)->count() > 0;
     }
+
+    /**
+     * @param int $fromProjectId
+     * @param int $toProjectId
+     * @return ProjectActivity[]
+     */
+    public function getDuplicatedActivityIds(int $fromProjectId, int $toProjectId): array
+    {
+        $q = $this->createQueryBuilder(ProjectActivity::class, 'activity');
+        $q->andWhere(
+            $q->expr()->orX(
+                $q->expr()->eq('activity.project', ':fromProjectId'),
+                $q->expr()->eq('activity.project', ':toProjectId')
+            )
+        )
+            ->setParameter('fromProjectId', $fromProjectId)
+            ->setParameter('toProjectId', $toProjectId)
+            ->andWhere('activity.deleted = :deleted')
+            ->setParameter('deleted', false);
+        $q->groupBy('activity.name')
+            ->having('counter >= 2')
+            ->select('activity, COUNT(activity.id) AS HIDDEN counter');
+        return $q->getQuery()->execute();
+    }
+
+    /**
+     * @param int $toProjectId
+     * @param array $fromProjectActivityIds
+     * @return array
+     */
+    public function saveCopyActivity(int $toProjectId, array $fromProjectActivityIds): array
+    {
+        // this will get all activities which belongs to $fromProjectActivityIds
+        $q = $this->createQueryBuilder(ProjectActivity::class, 'activity');
+        $q->andWhere($q->expr()->in('activity.id', ':fromProjectActivityIds'))
+            ->setParameter('fromProjectActivityIds', $fromProjectActivityIds);
+
+        /** @var ProjectActivity[] $fromProjectActivities */
+        $fromProjectActivities = $q->getQuery()->execute();
+
+        foreach ($fromProjectActivities as $fromProjectActivity) {
+            $toProjectActivity = new ProjectActivity();
+            $toProjectActivity->getDecorator()->setProjectById($toProjectId);
+            $toProjectActivity->setName($fromProjectActivity->getName());
+
+            $this->getEntityManager()->persist($toProjectActivity);
+        }
+        $this->getEntityManager()->flush();
+        return $fromProjectActivityIds;
+    }
```

Comment:
what do you think on this?
```suggestion
    /**
     * @param int $toProjectId
     * @param int[] $fromProjectActivityIds
     */
    public function saveCopyActivity(int $toProjectId, array $fromProjectActivityIds): void
    {
        // this will get all activities which belongs to $fromProjectActivityIds
        $q = $this->createQueryBuilder(ProjectActivity::class, 'activity');
        $q->andWhere($q->expr()->in('activity.id', ':fromProjectActivityIds'))
            ->setParameter('fromProjectActivityIds', $fromProjectActivityIds);

        /** @var ProjectActivity[] $fromProjectActivities */
        $fromProjectActivities = $q->getQuery()->execute();

        foreach ($fromProjectActivities as $fromProjectActivity) {
            $toProjectActivity = new ProjectActivity();
            $toProjectActivity->getDecorator()->setProjectById($toProjectId);
            $toProjectActivity->setName($fromProjectActivity->getName());

            $this->getEntityManager()->persist($toProjectActivity);
        }
        $this->getEntityManager()->flush();
    }
```

Reply:
done